articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,362 @@
+
+### Step 3: Project Structure
+
+All the files that we will be working with are located in the `src` folder. These files will be compiled and shown to the user.
+
+The image below shows the overall structure of the React application.
+
+![structure](/engineering-education/how-to-consume-data-from-an-API-using-react/structure.png)
+
+### Step 4: Developing the react application
+
+Use `npm start` to start the development server. By default, the react projects are usually hosted at `http://localhost:3000/`.
+
+We'll write all of our code in the `App.js`. Modify the `App.js` file to look like this.
+
+```javascript
+function App() {
+  return (
+    <div className="App">
+    
+    </div>
+  );
+}
+export default App;
+```
+
+Please note that this component will hold our logic for requesting and handling data from the OMDb API.
+
+We should import `useEffect` and `useState` into this component.
+
+```javascript
+import React, {useEffect, useState} from 'react';
+```
+
+The `useEffect` hook allows `React` to perform a specific action when the state changes. In our case, we will use `useEffect` to make network requests to the API when the component mounts for the first time.
+
+The `useState` hook will help in creating a state and a function to update that state. When you update a state, the component will re-render.
+
+Let's create a state for the movie. The `movie` state will hold data returned from the OMDb API. The `setMovie` helps in updating the `movie` state.
+
+```javascript
+const [movie, setMovie] = useState([]);
+```
+
+Next, Let's create a state to hold the search input from the user. This value is updated by the `setSearch` function whenever the user enters a value in the input field. The final value is then added to the search url.
+
+```javascript
+const [search, setSearch] = useState('');
+```
+
+You can also have a constant for your API key or include it in your url directly.
+
+```javascript
+const API_KEY = "your_api_key";
+const url = `http://www.omdbapi.com/?t=${search}&apikey=${API_KEY}`;
+```
+
+The `App` component also has several functions which are discussed below.
+
+#### 1.	getMovie
+
+This method will be called whenever the search button is clicked. It will make a request to the API and return a movie object. The `getMovie` function uses async and await. This is because network operations may take some time before they are completed. We, therefore, need to wait for the result. When we get the response, we convert it into JSON and store it in the movie variable. 
+
+Async - It helps in performing promise-based or asynchronous operations. 
+await -  keyword allows you to wait for a Promise.
+
+A promise is something which can be returned after the completion of a asynchronous operation.
+
+You can learn more about async, await, and promise from [here](https://www.w3schools.com/js/js_promise.asp).
```

Comment:
Can you explain more about Promises? You should talk about how it can either get resolved or rejected. Also a better explanation about what a Promise is required. Refer [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Promise is a very important part of API requests. Hence, a better explanation is necessary for the readers to fully understand the topic.